sdk/src/Core/Amazon.Runtime/Telemetry/Attributes.cs

```diff
@@ -0,0 +1,84 @@
+﻿/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ * 
+ *  http://aws.amazon.com/apache2.0
+ * 
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+using System.Collections.Generic;
+
+namespace Amazon.Runtime.Telemetry
+{
+    /// <summary>
+    /// Represents a collection of key-value pairs for attributes.
+    /// </summary>
+    public class Attributes
+    {
+        private readonly List<KeyValuePair<string, object>> _attributes;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Attributes"/> class.
+        /// </summary>
+        /// <param name="attributes">An initial collection of key-value pairs to populate the attributes.</param>
+        public Attributes(IEnumerable<KeyValuePair<string, object>> attributes)
+        {
+            _attributes = new List<KeyValuePair<string, object>>(attributes);
+        }
+
+        /// <summary>
+        /// Sets a key-value pair in the attributes collection.
+        /// </summary>
+        /// <param name="key">The key of the attribute.</param>
+        /// <param name="value">The value of the attribute.</param>
+        public void Set(string key, object value)
+        {
+            // Check if the key already exists
+            var index = _attributes.FindIndex(kvp => kvp.Key == key);
+            if (index != -1)
+            {
+                // Replace the existing key-value pair
+                _attributes[index] = new KeyValuePair<string, object>(key, value);
+            }
+            else
+            {
+                // Add a new key-value pair
+                _attributes.Add(new KeyValuePair<string, object>(key, value));
+            }
+        }
+
+        /// <summary>
+        /// Retrieves the value associated with the specified key.
+        /// </summary>
+        /// <param name="key">The key of the attribute to retrieve.</param>
+        /// <returns>The value associated with the key, or null if the key does not exist.</returns>
+        public object Get(string key)
+        {
+            // Check if the key already exists
+            var index = _attributes.FindIndex(kvp => kvp.Key == key);
+            if (index != -1)
+            {
+                // Replace the existing key-value pair
+                return _attributes[index];
```

Comment:
This should be `return _attributes[index].Value` per the method comments. There is also a problem with this method in general because the Value could be `null` which would indicate that the key wasn't found. Should `null` be allowed as a value?

Reply:
Fixed after using dictionary.